redis_heartbeat_lock/redis_heartbeat_lock.py

```diff
@@ -1 +1,46 @@
-"""Main module."""
+"""Main module. Builds on top of our redis client, adding a heartbeat and context manager."""
+
+import asyncio
+
+from .async_redis_lock import AsyncRedisLock
+
+
+class RedisHeartbeatLock:
+    """Manages starting and stopping heartbeat polling on a particular endpoint"""
+
+    # Period to poll the heartbeat on
+    period: float
+
+    # The heartbeat task
+    future: asyncio.Task
+
+    # Redis client
+    redis: AsyncRedisLock
+
+    def __init__(self, period: float, redis: AsyncRedisLock):
+        self.period = period
+        self.redis = redis
+
+    async def heartbeat(self) -> None:
+        """Refresh the Redis lock and go back to sleep."""
+        while True:
+            await self.redis.expire()
+            await asyncio.sleep(self.period)
+
+    async def __aenter__(self) -> None:
+        """Start the heartbeat. First, we set the Redis lock, then start the background task."""
+        lock = await self.redis.set_lock(value=True, nx=True)
```

Comment:
do you need to check the value of the key to see if the lock was stolen from you?